### PR TITLE
[Enhancement] Add FE debug tools: async-profiler

### PR DESCRIPTION
**Pull request by gengjun-git**
Generate memory profile
```
./bin/profiler.sh -e alloc -d 300 -f alloc-profile.html pid
```

Generate cpu profile
```
./bin/profiler.sh -d 300 -f cpu-profile.html pid
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by nshangyiming** [comment]
@kevincai  plz also help to review this pr

File: build.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ if [[ $OSTYPE == darwin* ]] ; then
     PARALLEL=$(sysctl -n hw.ncpu)
     # We know for sure that build-thirdparty.sh will fail on darwin platform, so just skip the step.
 else
-    if [[ ! -f ${STARROCKS_THIRDPARTY}/installed/include/datasketches/hll.hpp ]]; then
```
Comment (nshangyiming) [inline build.sh:49]: why replace the original condition, should use `||` operator?

File: thirdparty/vars.sh [hunk 2]
```diff
@@ -390,8 +390,15 @@ DATASKETCHES_NAME="datasketches-cpp-4.0.0.tar.gz"
 DATASKETCHES_SOURCE="datasketches-cpp-4.0.0"
 DATASKETCHES_MD5SUM="724cd1df9735de2b8939d298f0d95ea2"
 
+# async-profiler
+ASYNC_PROFILER_DOWNLOAD="http://starrocks-thirdparty.oss-cn-zhangjiakou.aliyuncs.com/async-profiler-2.9-linux-x64.tar.gz"
```
Comment (kevincai) [inline thirdparty/vars.sh:394]: will break arm build

File: build.sh [hunk 3]
```diff
@@ -46,7 +46,7 @@ if [[ $OSTYPE == darwin* ]] ; then
...
+    if [[ ! -f ${STARROCKS_THIRDPARTY}/installed/async-profiler/profiler.sh ]]; then
```
Comment (kevincai) [inline build.sh:49]: this is not a must have change to the compiling, better not change the condition to disturb everyone for a thirdparty rebuild.

File: thirdparty/vars.sh (hunk 2 above)
Comment (kevincai) [reply thirdparty/vars.sh:394]: suggest using github link, so maintainer can still find the original location and update if necessary.

https://github.com/async-profiler/async-profiler/releases

Comment (gengjun-git) [reply thirdparty/vars.sh]: done

File: build.sh (hunk 3 above)
Comment (gengjun-git) [reply build.sh]: done

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Astralidea — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**